frontend/src/features/env/EnvService.ts

```diff
@@ -23,7 +26,25 @@ const createFeedbackResponsesArray = (
 ) => {
   return feedbackFormFieldsStructure.map((question, i) => {
     const { _id, fieldType } = feedbackForm.form.form_fields[i]
-    const answer: string = formInputs[question] ?? ''
+    if (fieldType === BasicField.Checkbox) {
+      const answerArray: string | string[] = formInputs[question] ?? []
+      if (formInputs[othersInputName] && Array.isArray(answerArray)) {
+        if (formInputs[question] === '[""]') {
+          // Remove case where non-Others checkbox is selected
+          answerArray.pop()
+        }
```

Comment:
Apologies, this code confuses me 😢 

`answerArray` is defined from `formInputs[question]`, so they have the same value; unless `formInputs[question]` is `undefined` (in which case `answerArray` would be an empty array).

So, when `answerArray` is an array (per the conditional being checked), then either `formInputs[question]` was an array, or `formInputs[question]` was `undefined`. In both cases `formInputs[question]` can never be equal to the string `[""]`, and the nested conditional block never runs, correct?

Could you help me understand how this works / is supposed to work? 😅

Reply:
sorry this was so confusing! I've made changes to the conditional and added more comments here [ce16502](https://github.com/opengovsg/FormSG/pull/5582/commits/ce1650222cf66faacd5dcbdab9d5cbb5e074d27d), hope it makes sense now!!